astroquery/alma/core.py

```diff
@@ -8,28 +8,181 @@
 import tarfile
 import string
 import requests
+import warnings
 from pkg_resources import resource_filename
 from bs4 import BeautifulSoup
+import pyvo
 
 from six.moves.urllib_parse import urljoin
 import six
 from astropy.table import Table, Column, vstack as table_vstack
 from astropy import log
 from astropy.utils.console import ProgressBar
+from astropy.utils.exceptions import AstropyDeprecationWarning
 from astropy import units as u
-import astropy.coordinates as coord
-import astropy.io.votable as votable
+from astropy.time import Time
 
-from ..exceptions import (RemoteServiceError, TableParseError,
-                          InvalidQueryError, LoginError)
+from ..exceptions import (RemoteServiceError, LoginError)
 from ..utils import commons
 from ..utils.process_asyncs import async_to_sync
 from ..query import QueryWithLogin
+from .tapsql import _gen_pos_sql, _gen_str_sql, _gen_numeric_sql,\
+    _gen_band_list_sql, _gen_datetime_sql, _gen_pol_sql, _gen_pub_sql,\
+    _gen_science_sql, _gen_spec_res_sql, ALMA_DATE_FORMAT
 from . import conf, auth_urls
 
+__all__ = {'AlmaClass', 'ALMA_BANDS'}
+
 __doctest_skip__ = ['AlmaClass.*']
 
 
+ALMA_TAP_PATH = 'tap'
+ALMA_SIA_PATH = 'sia'
+
+# Map from ALMA ObsCore result to ALMA original query result
+# The map is provided in order to preserve the name of the columns in the
+# original ALMA query original results and make it backwards compatible
+# key - current column, value - original column name
+_OBSCORE_TO_ALMARESULT = {
+    'proposal_id': 'Project code',
+    'target_name': 'Source name',
+    's_ra': 'RA',
+    's_dec': 'Dec',
+    'gal_longitude': 'Galactic longitude',
+    'gal_latitude': 'Galactic latitude',
+    'band_list': 'Band',
+    's_region': 'Spatial resolution',
+    'em_resolution': 'Frequency resolution',
+    'antenna_arrays': 'Array',
+    'is_mosaic': 'Mosaic',
+    't_exptime': 'Integration',
+    'obs_release_date': 'Release date',
+    'frequency': 'Frequency support',
```

Comment:
I changed this & coi_name, but that triggered some remote test failures.  My changes may therefore be incorrect and need to be reverted - feel free to do so, but let's make sure the docs examples work.